driver/csiplugin/gpfs_util.go

```diff
@@ -390,3 +424,76 @@ func numberInSlice(a int, list []int) bool {
 	}
 	return false
 }
+
+func getVolIDMembers(vID string) (scaleVolId, error) {
+	splitVid := strings.Split(vID, ";")
+	var vIdMem scaleVolId
+
+	if len(splitVid) == 3 {
+		/* This is LW volume */
+		/* <cluster_id>;<filesystem_uuid>;path=<symlink_path> */
+		vIdMem.ClusterId = splitVid[0]
+		vIdMem.FsUUID = splitVid[1]
+		SlnkPart := splitVid[2]
+		slnkSplit := strings.Split(SlnkPart, "=")
+		if len(slnkSplit) < 2 {
+			return scaleVolId{}, status.Error(codes.Internal, fmt.Sprintf("Invalid Volume Id : [%v]", vID))
+		}
+		vIdMem.SymLnkPath = slnkSplit[1]
+		vIdMem.IsFilesetBased = false
+		return vIdMem, nil
+	}
+
+	if len(splitVid) == 4 {
+		/* This is fileset Based volume */
+		/* <cluster_id>;<filesystem_uuid>;fileset=<fileset_id>;path=<symlink_path> */
+		vIdMem.ClusterId = splitVid[0]
+		vIdMem.FsUUID = splitVid[1]
+		fileSetPart := splitVid[2]
+		fileSetSplit := strings.Split(fileSetPart, "=")
+		if len(fileSetSplit) < 2 {
+			return scaleVolId{}, status.Error(codes.Internal, fmt.Sprintf("Invalid Volume Id : [%v]", vID))
+		}
+
+		if fileSetSplit[0] == "filesetName" {
+			vIdMem.FsetName = fileSetSplit[1]
+		} else {
+			vIdMem.FsetId = fileSetSplit[1]
+		}
+
+		SlnkPart := splitVid[3]
+		slnkSplit := strings.Split(SlnkPart, "=")
+		if len(slnkSplit) < 2 {
+			return scaleVolId{}, status.Error(codes.Internal, fmt.Sprintf("Invalid Volume Id : [%v]", vID))
+		}
+		vIdMem.SymLnkPath = slnkSplit[1]
+		vIdMem.IsFilesetBased = true
+		return vIdMem, nil
+	}
+
+	if len(splitVid) == 8 {
+		/* Volume ID created from 2.5.0 onwards  */
+		/* VolID: <storageclass_type>;<type_of_volume>;<cluster_id>;<filesystem_uuid>;<consistency_group>;<application>;<fileset_name>;<symlink_path> */
+		vIdMem.StorageClassType = splitVid[0]
+		vIdMem.VolType = splitVid[1]
+		vIdMem.ClusterId = splitVid[2]
+		vIdMem.FsUUID = splitVid[3]
+		vIdMem.ConsistencyGroup = splitVid[4]
+		vIdMem.ApplicationName = splitVid[5]
+		vIdMem.FsetName = splitVid[6]
+		if vIdMem.StorageClassType == STORAGECLASS_CLASSIC {
+			if vIdMem.VolType == FILE_DIRECTORYBASED_VOLUME {
+				vIdMem.IsFilesetBased = false
+			} else {
+				vIdMem.IsFilesetBased = true
+			}
+		} else {
+			vIdMem.IsFilesetBased = true
```

Comment:
why not have check for volume type when storage class in not classic ?

Reply:
because we have only one type of volume for advanced storage class.